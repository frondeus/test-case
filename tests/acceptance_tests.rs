#![cfg(test)]

mod acceptance {
    use insta::with_settings;
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::{Command, Output};

    fn get_snapshot_directory() -> String {
        PathBuf::from("snapshots")
            .join(env::var("SNAPSHOT_DIR").unwrap_or_else(|_| "rust-stable".to_string()))
            .to_str()
            .unwrap()
            .to_string()
    }

    fn retrieve_stdout(output: &Output) -> String {
        String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .filter(|s| !s.is_empty())
            .map(|line| match line.find("; finished in") {
                Some(idx) => &line[0..idx],
                None => line,
            })
            .sorted()
            .join("\n")
    }

    #[test]
    fn basic() {
        with_settings!({snapshot_path => get_snapshot_directory()}, {
            let output = Command::new("cargo")
                .current_dir(PathBuf::from("acceptance_tests").join("basic"))
                .args(&["test"])
                .output()
                .expect("cargo command failed to start");

            let lines = retrieve_stdout(&output);
            insta::assert_display_snapshot!(lines);
        });
    }

    #[test]
    fn async_tests() {
        with_settings!({snapshot_path => get_snapshot_directory()}, {
            let output = Command::new("cargo")
                .current_dir(PathBuf::from("acceptance_tests").join("async_tests"))
                .args(&["test"])
                .output()
                .expect("cargo command failed to start");

            let lines = retrieve_stdout(&output);
            insta::assert_display_snapshot!(lines);
        });
    }

    #[test]
    fn test_item_reuse() {
        with_settings!({snapshot_path => get_snapshot_directory()}, {
            let output = Command::new("cargo")
                .current_dir(PathBuf::from("acceptance_tests").join("test_item_reuse"))
                .args(&["test"])
                .output()
                .expect("cargo command failed to start");

            let lines = retrieve_stdout(&output);
            insta::assert_display_snapshot!(lines);
        });
    }

    #[test]
    fn test_item_reuse_build() {
        with_settings!({snapshot_path => get_snapshot_directory()}, {
            let status = Command::new("cargo")
                .current_dir(PathBuf::from("acceptance_tests").join("test_item_reuse"))
                .args(&["build"])
                .status()
                .expect("cargo command failed to start");

            assert!(status.success())
        });
    }
}
