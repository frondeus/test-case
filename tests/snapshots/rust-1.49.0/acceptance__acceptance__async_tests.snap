---
source: tests/acceptance_tests.rs
assertion_line: 55
expression: lines

---
running 2 tests
test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
test test_cases::works_seamlessly_with_async_std::_100i32_expects_100usize ... ok
test test_cases::works_seamlessly_with_tokio::_100i32_expects_100usize ... ok
